sips/sips/sip-009-nft-standard.md

```diff
@@ -0,0 +1,93 @@
+
+NFTs are enumerated, the id starts at 1 and the current last id is provided by the smart contract.
+
+# Specification
+
+NFT smart contract shall implement the trait defined at `ST314JC8J24YWNVAEJJHQXS5Q4S9DX1FW5Z9DK9NT.nft-trait` as well as satisfy the additional conditions.
+The trait has three functions:
+* `last-token-id` does not take any arguments and returns the highest number that is used as an identifier for any NFT. This is the upper limit when iterating through all NFTs.
+* `get-owner` takes an NFT identifier and returns a response containing the principal owning the NFT for the given identifier. The principal is wrapped as an optional, that means if the corresponding NFT does not exists the response is `(ok none)`, otherwise, e.g. `(ok (some 'ST12...))`. The owner can be a contract principal.
+* `transfer` takes an NFT identifier, a sender principal and a receiver principal. The function changes the ownership of the NFT for the given identifier. The change has to be reflected in the `get-owner` function, for details see implementation rules.
+
+## Trait
+
+```
+(define-trait stacks-token-nft-standard-v1
+  (
+    ;; Token ID, limited to uint range
+    (last-token-id () (response uint uint))
+
+    ;; Owner of given token identifier
+    (get-owner (uint) (response (optional principal) uint))
+
+    ;; Transfer from to
+    (transfer (uint principal principal) (response bool (tuple (kind (string-ascii 32)) (code uint))))
+  )
+)
+```
+
+## Implementation rules
+
+1. Contracts must use a least one NFT asset. A post condition with deny mode and without any NFT condition about a changed owner must fail for `transfer` function calls.
+1. After a successful call to function `transfer` the function `get-owner` must return the recipient of the `transfer` call as the new owner.
+1. If a call to function `get-owner` returns some principal `A` value then it must return the same value until `transfer` is called with principal `A` as a sender
+1. For any call to `get-owner`, resp. `transfer` with an id greater than `last-token-id`, the call should return a response `none`, resp. failed transfer. 
+1. The following error codes are defined
+
+| function | error | description |
+|----------|-------|-------------| 
+|`transfer`|`{kind: "nft-transfer-failed", code: from-nft-transfer}`| Error if the call failed due to the underlying asset transfer. The code `from-nft-transfer` is the error code from the native asset transfer function|
+
+# Related Work
+
+https://eips.ethereum.org/EIPS/eip-721
+https://www.ledger.com/academy/what-are-nft
+
+# Backwards Compatibility
+
+Not applicable
+
+# Activation
+
+This SIP is activated if 5 contracts are deployed that are using the same trait that follows this specification. This has to happen before Stacks tip #5000.
```

Comment:
This block height has already passed.  Want to just set this to something really far into the future, like Bitcoin block 700,000, the way @hstove did for SIP 010?